mobile-widgets/qmlmanager.cpp

```diff
@@ -323,6 +324,10 @@ QMLManager::QMLManager() : m_locationServiceEnabled(false),
 	connect(Command::getUndoStack(), &QUndoStack::undoTextChanged, this, &QMLManager::undoTextChanged);
 	connect(Command::getUndoStack(), &QUndoStack::redoTextChanged, this, &QMLManager::redoTextChanged);
 
+  // Add connections to monitor for changes to the Deco settings
+	connect(qPrefTechnicalDetails::instance(), &qPrefTechnicalDetails::gflowChanged, this, &QMLManager::gflowChanged);
+	connect(qPrefTechnicalDetails::instance(), &qPrefTechnicalDetails::gfhighChanged, this, &QMLManager::gfhighChanged);
```

Comment:
I'm not convinced that this is the best way to do this - but I'd need to do more digging in the code.
Look at this... https://github.com/subsurface/subsurface/blob/25cfd852b910c7bbd53f0a1a5244c659761c2cc5/core/settings/qPrefTechnicalDetails.cpp#L60-L78
doesn't that seem to imply that `set_gf()` already gets called?

Reply:
I can only find disk_gfhigh() called in two places -- once in qPrefTechnicalDetails::loadSync and once in qPrefTechnicalDetails::set_gfhigh which you included above. 

set_gfhigh() is the method called when the preference is updated either on the mobile app or the desktop app. In that case as you can see, disk_gfhigh is called with doSync=true which means from the if(doSync) statement that set_gf does not get called, since it is only in the "else" code block. I was not sure of the implications of moving the set_gf outside the if/then/else block since it would impact both the desktop and mobile versions, so I looked for another solution that I could limit to only the mobile app.

What I did was to catch the gfhighChanged signal that is emitted on the next line after the call to disk_gf(true) in the qmlmanager.cpp code since that is limited to the mobile app.

Reply:
Also, as I mentioned in our other email thread, on the desktop app set_gf() is called in the desktop specific code in desktop-widgets/preferences/preferences-graph.cpp line 71 after the preferences are updated.

Reply:
I've now looked at core/settings/README.md and it looks like the doSync variable is meant to distinguish between when the preference are loaded at startup (doSync==false) or sync'd to save all settings to disk (doSync=true). It appears that currently set_gf() is only called when the preferences are loaded, but it should be possible to also call it when syncing changed values for GFLow and GFHigh to disk and eliminate the call from preferences_graph.cpp and from the addition I made to qmlmanager.cpp.

Can you think of any downside to setting the GFs in the Buhlmann algorithm each time they are sync'd to the disk?

Reply:
I moved the set_gf call to the set_gfhigh and set_gflow methods to cover both desktop and mobile apps and force pushed the commit. It appears everything still works as expected and it did not impact the planner in any way on the desktop app as far as I can tell. It might be good to have another set of eyes look at it though.

Reply:
I think this is the right direction. I was reading through the code and got distracted a few times (day job getting in the way) - but yes, I don't know why the `set_gf()` call would only be in the `doSync == false` case. That doesn't seem to make any sense. Why would you want to set the GF and then not want it applied?
@atdotde -- any comment on this?